dev/cuda/encoder_backward.cu

```diff
@@ -84,6 +89,29 @@ __global__ void encoder_backward_kernel2(float* dwte, float* dwpe,
     }
 }
 
+// Optimized implementation using shared memory
+__global__ void encoder_backward_kernel3(float* dwte, float* dwpe,
+                                         const float* dout, const int* inp,
+                                         int B, int T, int C) {
+    extern __shared__ float shared_mem[];
```

Comment:
wait this is not used right?

Reply:
No, I added that on my own rather than from the actual GPT code.